From d171dec94b6dfb1210d07bbf298eb33cce6d1eb2 Mon Sep 17 00:00:00 2001
From: Michal Simek <monstr@monstr.eu>
Date: Tue, 3 Apr 2012 13:51:20 +0200
Subject: [PATCH] remoteproc: zynq: Release declared coherent memory

Release memory to be be able to load module again.
There is still problem with cpu1 core reset.

Signed-off-by: Michal Simek <monstr@monstr.eu>
---
 drivers/remoteproc/zynq_remoteproc.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/drivers/remoteproc/zynq_remoteproc.c b/drivers/remoteproc/zynq_remoteproc.c
index 6cad2d3..b3994a7 100644
--- a/drivers/remoteproc/zynq_remoteproc.c
+++ b/drivers/remoteproc/zynq_remoteproc.c
@@ -275,6 +275,8 @@ static int __devexit zynq_remoteproc_remove(struct platform_device *pdev)
 
 	dev_info(&pdev->dev, "%s\n", __func__);
 
+	dma_release_declared_memory(&pdev->dev);
+
 	clear_ipi_handler(local->ipino);
 	clear_irq(pdev);
 
-- 
1.7.0.4

